samples/nextjs/scripts/templates/component-factory.ts

```diff
@@ -0,0 +1,26 @@
+/**
+ * Generates the contents of the component factory file using a predefined string template.
+ * @param imports - the list of import statements
+ * @param registrations - the list of component name to component type mappings
+ * @returns component factory file contents
+ */
+function generateComponentFactory(imports: string[], registrations: string[]): string {
+  return `/* eslint-disable */
+// Do not edit this file, it is auto-generated at build time!
+// See scripts/generate-component-factory.ts to modify the generation of this file.
+${imports.join('\n')}
+
+const components = new Map();
+${registrations.join('\n')}
```

Comment:
There's a template dependency baked into the `registrations` array: the **components** in `components.set(...);`. 
Would it be cleaner to pass this `generateComponentFactory` the component list (`ComponentFile[]`) instead and let it be responsible for building those import/registrations internally? That way the generated code, including the "component", all live together in this file.

Reply:
good idea